components/server/src/authorization/authorizer.ts

```diff
@@ -157,14 +160,26 @@ export class Authorizer {
     }
 
     async addUser(userId: string, owningOrgId?: string) {
-        await this.authorizer.writeRelationships(
-            set(rel.user(userId).self.user(userId)), //
-            set(
-                owningOrgId
-                    ? rel.user(userId).organization.organization(owningOrgId)
-                    : rel.user(userId).installation.installation,
-            ),
+        const oldOrgs = await this.findAll(rel.user(userId).organization.organization(""));
+        const updates = [set(rel.user(userId).self.user(userId))];
+        updates.push(
```

Comment:
I would feel waaay better if we had a dedicated set-based set of tests for this logic. :orange_heart:
In theory it's simple, but I have to re-read multiple times to grok these lines. :slightly_smiling_face:

Reply:
This [test](https://github.com/gitpod-io/gitpod/pull/18394/files#diff-433881f2017ff6d513adf106be976c3df02126c6b2353b3c78d5a479e02cfbfeR62-R97) does that. The relationship updater itself doesn't add much logic to it, but yes we can also have a test suite for authorizer itself. 👍

Reply:
I saw those. But it would be to have more synthetic tests on the authorizer, that test all combinations we're after here.